keep-runtime/src/main.rs

```diff
@@ -18,23 +18,167 @@
+        .run(listen_socketaddr)
+        .await;
+}
+
+fn create_new_runtime(recvd_data: &[u8]) {
+    format!("About to attempt new runtime creation");
+    let _ = env_logger::try_init_from_env(env_logger::Env::default());
+    //TODO - get args these from main() if required
+    //    let args = std::env::args().skip(1);
+    let dummy_arr: [&str; 1] = [""];
+    let vars = std::env::vars();
 
+    let result = workload::run(recvd_data, &dummy_arr, vars).expect("Failed to run workload");
+    println!("Got result (println) {:#?}", result);
     info!("got result: {:#?}", result);
 }
+
+async fn payload_launch(payload: Payload) -> Result<impl warp::Reply, warp::Rejection> {
+    format!("Received a {} file", payload.encoding);
+    println!("Received a {} file", payload.encoding);
+    create_new_runtime(&payload.contents);
+    Ok(warp::reply::with_status(
+        "Payload received",
+        warp::http::StatusCode::OK,
+    ))
+}
+
+fn get_credentials_bytes(listen_addr: &str) -> (Vec<u8>, Vec<u8>) {
+    let (key, cert) = match KEY_SOURCE {
+        "file-system" => (get_key_bytes_fs(), get_cert_bytes_fs()),
+        "generate" => (generate_credentials(&listen_addr)),
+        //no match!
+        _ => panic!("No match for credentials source"),
+    };
+    (key, cert)
+}
+
+//implementation for file system
+fn get_cert_bytes_fs() -> Vec<u8> {
+    let in_path = Path::new("key-material/server.crt");
+
+    let in_contents = match std::fs::read(in_path) {
+        Ok(in_contents) => {
+            println!("Contents = of {} bytes", &in_contents.len());
+            in_contents
+        }
+        Err(_) => {
+            println!("Failed to read from file");
+            panic!("We have no data to use");
+        }
+    };
+    in_contents
+}
+
+//implementation for file system
+fn get_key_bytes_fs() -> Vec<u8> {
+    println!("Generating server key (PEM)");
+    let in_path = Path::new("key-material/server.key");
+
+    let in_contents = match std::fs::read(in_path) {
+        Ok(in_contents) => {
+            println!("Contents = of {} bytes", &in_contents.len());
+            in_contents
+        }
+        Err(_) => {
+            println!("Failed to read from file");
+            panic!("We have no data to use");
+        }
+    };
+    in_contents
+}
+
+//TODO - this is vital code, and needs to be carefully audited!
+fn generate_credentials(listen_addr: &str) -> (Vec<u8>, Vec<u8>) {
+    let key = Rsa::generate(2048).unwrap();
+    let pkey = PKey::from_rsa(key.clone()).unwrap();
+
+    println!(
+        "Should create a certificate for {}, but using hard-coded 127.0.0.1 instead",
+        &listen_addr
+    );
+
+    let mut x509_name = openssl::x509::X509NameBuilder::new().unwrap();
+    x509_name.append_entry_by_text("C", "GB").unwrap();
+    x509_name.append_entry_by_text("O", "enarx-test").unwrap();
+    //FIXME - we should use &listen-addr, but this fails
+    //x509_name.append_entry_by_text("CN", &listen_addr).unwrap();
```

Comment:
This works for me... what's the problematic listen_addr which does not work?

Reply:
When running the app-loader-tester against this (on a "real" IP address), it fails to validate the cert.